tpl/template_funcs_test.go

```diff
@@ -324,6 +324,30 @@ func TestAfter(t *testing.T) {
 	}
 }
 
+func TestDictionary(t *testing.T) {
+	for i, this := range []struct {
+		v1     []interface{}
+		expecterr bool
+		expectedValue map[string] interface{}
+	}{
+		{[]interface{}{"a", "b"}, false, map[string]interface{}{"a":"b"}},
+		{[]interface{}{5, "b"}, true,nil},
+		{[]interface{}{"a", 12,"b",[]int{1, 2, 4}}, false,map[string]interface{}{"a":12,"b":[]int{1, 2, 4}}},
+		{[]interface{}{"a", "b", "c"}, true,nil},
+	} {
+		r,e := Dictionary(this.v1...)
+
+		if (this.expecterr && e==nil) || (!this.expecterr && e!=nil)  {
+			t.Errorf("[%d] got an unexpected error", i, e, this.expecterr)
+		} else if !this.expecterr {
+			if !reflect.DeepEqual(r, this.expectedValue) {
```

Comment:
@bep This is the code in DeepEqual for comparing a map in deep equal it does *not* use the order of the keys to make the comparison, it iterates over the map keys and fetches the value per key. 

			for _, k := range v1.MapKeys() {
				if !deepValueEqual(v1.MapIndex(k), v2.MapIndex(k), visited, depth+1) {
					return false
				}
			}